pylint/message/message_handler_mix_in.py

```diff
@@ -267,14 +279,20 @@ def check_message_definition(message_definition, line, node):
                         f"Message {message_definition.msgid} must provide Node, got None"
                     )
 
-    def add_one_message(
-        self, message_definition, line, node, args, confidence, col_offset
-    ):
+    def add_one_message(  # type: ignore # MessagesHandlerMixIn is always mixed with PyLinter
+        self: "PyLinter",
+        message_definition: "MessageDefinition",
+        line: Optional[int],
+        node: Optional[nodes.NodeNG],
+        args: Union[str, Tuple[Union[str, int], ...], None],
```

Comment:
Is it really limited to only these three types? This is the signature for `str.format`:
```py
def format(self, *args: object, **kwargs: object) -> str: ...
```

Reply:
Yes, I tested this and `mypy` complains whenever a call has other types for `args`. So I think we cans safely assume this is correct.

Reply:
You just need to replace all reference, then it should work.
* `BaseChecker.add_message`
* `MessagesHandlerMixin.add_message`
* `MessagesHandlerMixin.add_one_message`

I would also recommend to use `args: Any` for now. We can always narrow it down later, if we choose to.

Reply:
I don't really understand the first half of your comment. Doesn't this PR already add the `args` typing to all those three places?

I think we can add this typing. `mypy` checks all calls to any of the three functions and checks whether `args` has the correct typing. This was the reason I added `int` to the `Tuple`, `mypy` notified me of that possibility. Perhaps `mypy` is not doing what I think it is, but I believe there is no way for `args` to be anything other than what is now specified in the typing without `mypy` telling us about it.

Reply:
That would be the case if the codebase was fully type. However `mypy` doesn't check the body of untyped function definitions. So there is the likely case that `Union[str, Tuple[Union[str, int], ...], None]` doesn't cover it.

Additionally, this only covers the our code but what about plugins? Can they for example pass a `float` with `args`.

Thus, I would recommend to replace it with `Any` (at the very least for now).

Reply:
I think #4940 made it so all function scopes which contain an `add_message` are typed. But you are right, there is a real possibility that plugins pass other types of `args`. I will change the typing to `Any`.

Reply:
Sadly, we aren't quite there yet. Adding types there also won't be that easy as it was for #4940. This one here took me < 1min to find:
https://github.com/PyCQA/pylint/blob/b3336c77378f88eb1428d6ff5b6e78bba1177f40/pylint/checkers/base.py#L689-L691